hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/SpawnBalancer.java

```diff
@@ -1256,26 +1243,18 @@ protected void updateAggregateStatistics(List<HostState> hosts) {
     }
 
     private HostScore calculateHostScore(HostState host, double clusterMaxMeanActive, double clusterMaxDiskUsed) {
-        double score = 0;
         double meanActive = host.getMeanActiveTasks();
-        double usedDiskPercent = getUsedDiskPercent(host);
-        // If either metric is zero across the whole cluster, treat every host as having full load in that aspect
-        if (clusterMaxMeanActive <= 0) {
-            meanActive = 1;
-            clusterMaxMeanActive = 1;
-        }
-        if (clusterMaxDiskUsed <= 0) {
-            usedDiskPercent = 1;
-            clusterMaxDiskUsed = 1;
-        }
+        double diskUsedPercentModified = host.getDiskUsedPercentModified(config.getMinFreeDiskSpaceToRunJobs());
         int activeTaskWeight = config.getActiveTaskWeight();
         int diskUsedWeight = config.getDiskUsedWeight();
         // Assemble the score as a combination of the mean active tasks and the disk used
-        score += activeTaskWeight * Math.pow(meanActive / clusterMaxMeanActive, 2.5);
-        score += diskUsedWeight * Math.pow(usedDiskPercent / clusterMaxDiskUsed, 2.5);
+        double exponent = 2.5;
+        double score = activeTaskWeight * Math.pow(meanActive, exponent);
+        double diskPercentPowered = Math.pow(diskUsedPercentModified, exponent);
+        score += diskUsedWeight * diskPercentPowered;
         // If host is very full, make sure to give the host a big score
-        score = Math.max(score, (activeTaskWeight + diskUsedWeight) * usedDiskPercent);
-        return new HostScore(meanActive, usedDiskPercent, score);
+        score = Math.max(score, (activeTaskWeight + diskUsedWeight) * diskPercentPowered);
```

Comment:
I still don't quite understand how it gives host with fuller disk a higher score.

The default configuration is: activeTaskWeight=30 and diskUsedWeight=70. meanActive is about 2 and doesn't seem to change a lot over time, according to ganglia metric. Now let's say disk is full, so diskUsedPercentModified=1

The original formula (before the Math.max call) produces: score=30*pow(2, 2.5) + 70*pow(1, 2.5)=170+70=240
The alternative formula (second parameter in the Math.max call) produces: score=(30+70)*pow(1, 2.5)=100

The original score still wins.

The thing is: disk percent is <=1 while mean active seems to be around 2. With the exponential, mean active is almost always going to be the dominant factor.

Where am I wrong in this?